client/layout/masterbar/publish.jsx

```diff
@@ -74,25 +98,17 @@ class MasterbarItemNew extends React.Component {
 		return (
 			<div className="masterbar__publish">
 				<MasterbarItem
-					ref={ this.setPostButtonContext }
+					ref={ this.setPostButtonRef }
 					url={ newPostPath }
 					icon="create"
 					onClick={ this.onClick }
 					isActive={ this.props.isActive }
 					tooltip={ this.props.tooltip }
 					className={ classes }
-					preloadSection={ () => preload( 'post-editor' ) }
+					preloadSection={ this.preloadPostEditor }
 				>
 					{ this.props.children }
-					<SitesPopover
-						id="popover__sites-popover-masterbar"
-						visible={ this.state.isShowingPopover }
-						context={ this.state.postButtonContext }
-						onClose={ this.toggleSitesPopover.bind( this, false ) }
-						onSiteSelect={ this.props.siteSelected }
-						groups={ true }
-						position={ this.getPopoverPosition() }
-					/>
+					{ this.renderPopover() }
```

Comment:
Late to the party and drive-by-ish, but I generally prefer `{ this.state.isShowingPopover && <SitesPopover /*...*/ />` over `this.renderSomething()` methods -- the latter just tend to accumulate and make things more abstract to read IMO.

Reply:
Yes, I agree that it can be hard to read the source of a class that has many `render*` methods. The best way how to structure the code and avoid too many `render*` methods is probably extracting a separate functional component?

In this case, my motivation was to separate the "`MasterbarItem` object with many props" from the "`SitePopover` component with many props" and not have too much nesting in the main `render` method.